From d5dc47fd8fe26d5601d57c603bb6769b33ef96b7 Mon Sep 17 00:00:00 2001
From: Mario Carrillo <mario.alfredo.c.arevalo@intel.com>
Date: Thu, 29 Oct 2015 18:35:38 -0600
Subject: [PATCH 03/10] mk: Add rule for installing libraries

Add hierarchy-file support to the DPDK libraries,
when invoking "make install-lib" libraries will
be installed in: $(DESTDIR)/$(RTE_PREFIX)/$(LIB_DIR)
where LIB_DIR=/usr/lib by default, you can override
RTE_PREFIX and LIB_DIR vars.
This hierarchy is based on:
http://www.freedesktop.org/software/systemd/man/file-hierarchy.html

Signed-off-by: Mario Carrillo <mario.alfredo.c.arevalo@intel.com>
---
 mk/rte.sdkinstall.mk | 16 ++++++++++++++++
 mk/rte.sdkroot.mk    |  4 ++--
 2 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/mk/rte.sdkinstall.mk b/mk/rte.sdkinstall.mk
index 9277dd2..58e5516 100644
--- a/mk/rte.sdkinstall.mk
+++ b/mk/rte.sdkinstall.mk
@@ -45,9 +45,11 @@ ifneq (,$(wildcard $(RTE_OUTPUT)/.config))
 RTE_PREFIX ?= /usr/local
 INCLUDE_DIR ?= $(RTE_PREFIX)/include/dpdk
 BIN_DIR ?= $(RTE_PREFIX)/bin
+LIB_DIR ?= $(RTE_PREFIX)/lib
 HSLINKS := $(wildcard $(RTE_OUTPUT)/include/*)
 BINARY_FILES := $(patsubst %.map,,$(wildcard $(RTE_OUTPUT)/app/*))
 NIC_FILES := $(wildcard $(RTE_SDK)/tools/*.py)
+LIBS := $(wildcard $(RTE_OUTPUT)/lib/*)
 BINARY_FILES += $(NIC_FILES)
 endif
 endif
@@ -111,6 +113,20 @@ install-bin:
 	done \
 	fi
 #
+# install libs in /usr/local/lib by default
+# RTE_PREFIX and LIB_DIR can be overridden.
+#
+.PHONY: install-lib
+install-lib:
+	@echo ================== Installing libraries
+	@if [ ! -z "${LIBS}" ]; then \
+	[ -d $(DESTDIR)/$(LIB_DIR) ] || mkdir -p $(DESTDIR)/$(LIB_DIR); \
+	for LIB in ${LIBS}; do \
+	cp -rf $$LIB ${DESTDIR}/${LIB_DIR}; \
+	echo installing: $$LIB; \
+	done \
+	fi
+#
 # uninstall: remove all built sdk
 #
 UNINSTALL_TARGETS := $(addsuffix _uninstall,\
diff --git a/mk/rte.sdkroot.mk b/mk/rte.sdkroot.mk
index 24eaa60..7a72c9b 100644
--- a/mk/rte.sdkroot.mk
+++ b/mk/rte.sdkroot.mk
@@ -97,8 +97,8 @@ test fast_test ring_test mempool_test perf_test coverage:
 testall:
 	$(Q)$(MAKE) -f $(RTE_SDK)/mk/rte.sdktestall.mk $@
 
-.PHONY: install install-headers install-bin uninstall
-install install-headers install-bin uninstall:
+.PHONY: install install-headers install-bin install-lib uninstall
+install install-headers install-bin install-lib uninstall:
 	$(Q)$(MAKE) -f $(RTE_SDK)/mk/rte.sdkinstall.mk $@
 
 .PHONY: doc help
-- 
2.6.1.120.g24a00ef

